dask_searchcv/tests/test_adaptive.py

```diff
@@ -0,0 +1,181 @@
+from dask_searchcv.adaptive import _top_k, Hyperband
+from dask_ml.linear_model import PartialSGDClassifier
+from dask_ml.datasets import make_classification
+import numpy as np
+import dask.array as da
+from dask.distributed import Client
+import pandas as pd
+import pytest
+import distributed
+from pprint import pprint
+import scipy.stats as stats
+import random
+from sklearn.linear_model import Lasso
+from distributed.utils_test import cluster, loop, gen_cluster
+import time
+from dask_ml.wrappers import Incremental
+
+
+class ConstantFunction:
+    def _fn(self):
+        return self.value
+
+    def get_params(self, deep=None, **kwargs):
+        return {k: getattr(self, k) for k, v in kwargs.items()}
+
+    def set_params(self, **kwargs):
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+        return self
+
+    def partial_fit(self, *args, **kwargs):
+        pass
+
+    def score(self, *args, **kwargs):
+        return self._fn()
+
+
+def _with_client(fn):
+    def fn_with_client(*args, **kwargs):
+        with cluster() as (s, [a, b]):
+            with Client(s['address'], loop=loop) as client:
+                y = fn(*args, **kwargs)
+        return y
+    return fn_with_client
+
+
+#  @_with_client
+@gen_cluster(client=True)
+def test_hyperband_sklearn(*args, **kwargs):
+    X, y = make_classification(n_samples=1000, chunks=500)
+    classes = da.unique(y)
+    model = Incremental(SGDClassifier(warm_start=True, classes=classes,
+                                      loss='hinge', penalty='elasticnet'))
+
+    params = {'alpha': np.logspace(-3, 0, num=int(10e3)),
+              'l1_ratio': np.linspace(0, 1, num=int(10e3))}
+    alg = Hyperband(model, params, max_iter=9, n_jobs=0)
+
+    alg.fit(X, y, dry_run=True)
+    assert len(alg.history) == 20
+    alg.fit(X, y, dry_run=True)
+    assert len(alg.history) == 40
+
+    alg.fit(X, y, classes=classes)
```

Comment:
The fit call calls `compute` internally, which blocks.  Ideally the following would work:

```python
delayed_object = alg.fit(X, y, classes=classes, compute=False)
future = c.compute(delayed_object)
result = yield future
```